### DISCUSSION TITLE
mintpy on WCape results

### DISCUSSION THREAD

**Discussion by MaryemBl** [insarlab/MintPy#800] [Q&A] [answered]
Hello everyone, 
I'm currently applying mintpy workflow on WCape dataset, link below, (i'm using snap), I've got results but cannot  validate them. I need expected results from using mintpy or other techniques (gps data...) applied on this area.
to anyone who have worked with this dataset, how did u validate your results? 
https://zenodo.org/record/4318134#.YrWhA3ZBxPY
Thank you

**Comment by mosyhey** [comment, marked as answer]
Hello,
As you know, the InSAR displacement time-series is relative in both time (i.e. reference date) and space (i.e. reference point). First, you should calibrate the displacement time-series with a CGPS station. Then, you can validate the calibrated displacement time-series with other CGPS stations.

I suggest this process, but I am not sure it's true. I hope @yunjunz, @hfattahi and other friends approve or correct it:

1. In step `reference_date`, set the reference date at the first date.
2. In step `reference_point`, set the location of a CGPS station as the reference point. Consider the following conditions in selecting the station:
```
## 1) located in a coherence area, to minimize the decorrelation effect.
## 2) not affected by strong atmospheric turbulence, i.e. ionospheric streaks
## 3) close to and with similar elevation as the AOI, to minimize the impact of spatially correlated atmospheric delay
```
3. Set other settings and run MintPy processing. Tropospheric correction is highly recommended.
4. After MintPy processing, calibrate the whole displacement time-series with the reference CGPS station data.
5. Eventually, validate the calibrated displacement time-series with the other CGPS stations.

You can see my primary validation result in our study area [here](https://github.com/insarlab/MintPy/issues/793#issuecomment-1159381469). I tried all sorts of settings, and got R2≈0.85 and RMSE≈8mm at best.

**Reply by yunjunz** [reply]
Your procedure looks reasonable to me.

**Comment by MaryemBl** [comment]
Thank u @mosyhey, @yunjunz for your answers, I will try this. In step 4, did u  include GNSS data in the InSAR 3D matrix to calibrate the displacement time-series with the reference CGPS station data?

**Comment by mosyhey** [comment]
Thank you dear @yunjunz.
Yes, dear @MaryemBl , I include the reference GNSS station data in InSAR matrix for calibration, in this procedure:

1. Doing steps 1 to 3 in the previous post. Imagine the resultant InSAR matrix has 3 layers or dates [Date1,Date2,Date3]. Also, imagine we selected CGPS A as the reference point for calibration and CGPS B for validation.
2. Getting data from CGPS stations on those three dates. If you access hourly or minute-by-minute data, get the data for the same time as image acquisition times.
3. GNSS data has three components: North, East, and Up (NEU). For example, component of U in CGPS A is [23,28,32]. Standardize every component of all stations by subtracting the first value from them. In this example, it will be [0,5,9].
4. Converting the CGPS NEU displacement to satellite LOS (Line of Sight) displacement, by [this formula](https://forum.step.esa.int/t/free-ground-validation-datasets-for-insar-gps-gnss-etc/30916/11). For example, LOS displacement of CGPS A calculated as [0,3,7].
5. Calibrating InSAR matrix [Date1,Date2,Date3] by adding CGPS A LOS displacement [0,3,7] to it (see image below).
6. Extracting calibrated InSAR LOS displacement value at the pixel of CGPS B. In this example, it is [0,5,11] (see image below).
7. Validating it [0,5,11] with CGPS B LOS displacement time-series (calculated in step 4) as [here](https://github.com/insarlab/MintPy/discussions/803#discussioncomment-3059855).

![insar_gps](https://user-images.githubusercontent.com/105506772/177693825-1a2baf60-cc0b-4e86-a08b-520241ad72f5.png)

Try different settings to get the best correlation. Please share the final R2 and RMSE.
